### PR TITLE
[component] make Type implement MarshalText

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** <Describe what has changed.>
Adds `MarshalYAML` function so that `Type` can be properly marshaled as yaml.

We could alternatively fix this issue by updating the `components` command to explicitly call `Type.String` and use that value instead of `component.Type` in its struct.

**Link to tracking Issue:** <Issue number if applicable>
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/9855

**Testing:** <Describe what testing was performed and which tests were added.>
Tested locally using `otelcorecol`.  The unit test `TestNewBuildSubCommand` didnt catch the issue because it was also using `yaml.Marshal` like the command, so the expected value also had `{}` as the name.

### CODE REVIEW

File: component/config.go [hunk 1]
```diff
@@ -119,6 +119,11 @@ func (t Type) String() string {
 	return t.name
 }
 
+// MarshalYAML marshals returns the Type name.
+func (t Type) MarshalYAML() (interface{}, error) {
```
Comment (dmitryax) [inline component/config.go:123]: I'm surprised we use yaml in that command. I think `MarshalText` should work as well. That would be consistent with the rest of the codebase

**Comment by dmitryax** [comment]
Also, the test for this functionality is pretty useless. It'd be great to update it to compare the expected output with an actual yaml string instead of doing the same marshaling. Could you please update it?

File: component/config.go (hunk 1 above)
Comment (TylerHelmuth) [reply component/config.go:123]: I just tried with `MarshalText` and it didn't work.  I can include both functions tho.

Comment (TylerHelmuth) [reply component/config.go:123]: NVM, it works if the implementation of `MarshalText` actually conforms to the interface lol

**Comment by TylerHelmuth** [comment]
I believe the windows test failure is unrelated

**Review by mx-psi — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]
Thanks, @TylerHelmuth!

**Pull request merged**